aspects/schema.go

```diff
@@ -740,15 +785,20 @@ type arraySchema struct {
 func (v *arraySchema) Validate(raw []byte) error {
 	var array *[]json.RawMessage
 	if err := json.Unmarshal(raw, &array); err != nil {
-		return err
+		return &validationError{baseErr: err}
```

Comment:
I never noticed but we never arrive here in the test?

Reply:
I think a few of these aren't tested because they're basically just testing if the JSON is malformed

Reply:
this is not really a malformed error though, it would happen also if we have number where we expect an array? no?

Reply:
True, malformed is incorrect, it's more unparseable as the expected type. I had created a ticket to inspect and wrap these errors so I've added a note there to re-check the coverage